tests/examples/form_data.rs

```diff
@@ -0,0 +1,73 @@
+use hyper::client::Client;
+use hyper::header::ContentType;
+
+use util::{read_body_to_string, read_url, run_example};
+
+static FORM : &'static str ="
+<!DOCTYPE html>
+<html>
+<head>
+    <meta http-equiv=\"Content-Type\" content=\"text/html\">
+    <title>Contact</title>
+</head>
+<body>
+      <form>
+            <input type=\"text\" name=\"firstname\" placeholder=\"First Name\" autofocus>
+            <input type=\"text\" name=\"lastname\" placeholder=\"Last Name\">
+            <input type=\"tel\" name=\"phone\" placeholder=\"Phone #\">
+            <input type=\"email\" name=\"email\" placeholder=\"Email\">
+            <button type=\"submit\" formenctype=\"application/x-www-form-urlencoded\" formaction=\"/confirmation\" formmethod=\"post\">SEND</button>
+      </form>
+</body>
+</html>
+";
+
+static CONFIRMATION : &'static str = "
+<!DOCTYPE html>
+<html>
+<head>
+    <meta http-equiv=\"Content-Type\" content=\"text/html\">
+    <title>Confirmation</title>
+</head>
+<body>
+    <h1>Confirmation</h1>
+    <dl>
+        <dt>First Name</dt>
+        <dd>John</dd>
+        <dt>Last Name</dt>
+        <dd>Doe</dd>
+        <dt>Phone</dt>
+        <dd>911</dd>
+        <dt>Email</dt>
+        <dd>john@doe.com</dd>
+    </dl>
+</body>
+</html>
+";
+
+#[test]
+fn form_data() {
+    run_example("form_data", |port| {
+        // Display form
+        let url = format!("http://localhost:{}/", port);
+        let s = read_url(&url);
+        assert_eq!(s.trim(), FORM.trim());
```

Comment:
Also changing the template from a definition list to a parameterized sentence would help for checking the `POST success` case.